065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,308 @@
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=3&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "8911ca89-351f-888-8d0f-9aade00e098h",
+      "StartMs": "1701961136014"
+    }
+  ],
+  "version": 1
+}
+```
+
+### Configuration parameters
+
+The Topic Operator will need to authenticate and send HTTP requests to the Cruise Control's REST API.
+A new admin user will be provisioned for the Topic Operator, separate from the credentials used by the KafkaRebalanceOperator.
+The following environment variables will be set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
+
+| Name                                  | Type    | Default | Description                                                   |
+|:--------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED        | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED   | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME       | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT           | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED    | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED   | Boolean | false   | Whether Basic authentication is enabled                       |
+
+With the Cluster Operator deployment, the certificate chain in PEM format and the admin user credentials for the Cruise Control REST API will be automatically mounted in the Topic Operator container.
+Instead, with the standalone deployment, the user will have to mount: certificate chain in `/etc/tls-sidecar/cluster-ca-certs/ca.crt`, credentials in `/etc/eto-cc-api/topic-operator.apiAdminName` and `/etc/eto-cc-api/topic-operator.apiAdminPassword`.
+
+### KafkaTopic CRD changes
+
+During the whole replicas change process the KafkaTopic will remain in a ready state.
+A replicas change will be executed asynchronously, taking more than one reconciliation to complete.
+
+A new optional KafkaTopicStatus subsection called `replicasChange` will be used to update the user, and keep track of the replicas change state.
+The `replicasChange` can be in a pending or ongoing `state`, and it can contain a `message` in case of failure.
+Other fields are `sessionId`, which maps to Cruise Control's `User-Task-ID`, and `targetReplicas`, which reflects the target replicas value (this may be different from .spec.replicas when the state is ongoing).
```

Comment:
```suggestion
Other fields are `sessionId`, which maps to Cruise Control's `User-Task-ID`, and `targetReplicas`, which reflects the target replicas value (this may be different from `.spec.replicas` when the state is ongoing).
```